tests/serde.rs

```diff
@@ -0,0 +1,16 @@
+#![cfg(feature = "serde")]
+#[macro_use]
+extern crate ego_tree;
+use std::assert_eq;
+
+use ego_tree::Tree;
+
+#[test]
+fn test_serialize() {
+    let tree = tree!("r" => {"a", "b" => { "d", "e" }, "c"});
+
+    let serialized = serde_json::to_string(&tree).unwrap();
+    let deserialized: Tree<&str> = serde_json::from_str(&serialized).unwrap();
```

Comment:
While a round trip test is fine (actually a very good candidate for a property-based test), I think having a test (using `serde_test`) which asserts the shape of the translation into the Serde data model itself would also be useful.

Reply:
When we reach a common decision on the format to use for serialization I will proceed to replace/integrate this test with something like this